src/client-side-metrics/exporter.ts

```diff
@@ -0,0 +1,409 @@
+  };
+  for (const scopeMetrics of exportArgs.scopeMetrics) {
+    for (const scopeMetric of scopeMetrics.metrics) {
+      const metricName = scopeMetric.descriptor.name;
+      for (const dataPoint of scopeMetric.dataPoints) {
+        const value = dataPoint.value;
+        const allAttributes = dataPoint.attributes;
+        const streaming = allAttributes.streamingOperation;
+        /*
+        metricLabels are built from the open telemetry attributes that are set
+        when a data point is recorded. This means that for one metric there may
+        be multiple time series' with different attributes, but the resource
+        labels will always be the same for a particular export call.
+         */
+        const metricLabels = Object.assign(
+          {
+            app_profile: allAttributes.appProfileId,
+            client_name: allAttributes.clientName,
+            method: allAttributes.methodName,
+            status:
+              (allAttributes as OnAttemptAttribute).attemptStatus?.toString() ??
+              (
+                allAttributes as OnOperationAttribute
+              ).finalOperationStatus?.toString(),
+            client_uid: allAttributes.clientUid,
+          },
+          streaming ? {streaming} : null
+        );
+        const metric = {
+          type: metricName,
+          labels: metricLabels,
+        };
+        const resource = {
+          type: exportArgs.resource._syncAttributes['monitored_resource.type'],
+          labels: resourceLabels,
+        };
+        const interval = {
+          endTime: {
+            seconds: dataPoint.endTime[0],
+          },
+          startTime: {
+            seconds: dataPoint.startTime[0],
+          },
+        };
+        const timeSeries = isCounterValue(value)
+          ? {
+              metric,
+              resource,
+              valueType: 'INT64',
+              points: [
+                {
+                  interval,
+                  value: {
+                    int64Value: dataPoint.value,
+                  },
+                },
+              ],
+            }
+          : {
+              metric,
+              resource,
+              metricKind: 'CUMULATIVE',
+              valueType: 'DISTRIBUTION',
+              points: [
+                {
+                  interval,
+                  value: {
+                    distributionValue: {
+                      count: String(value.count),
+                      mean: value.count ? value.sum / value.count : 0,
+                      bucketOptions: {
+                        explicitBuckets: {
+                          bounds: value.buckets.boundaries,
+                        },
+                      },
+                      bucketCounts: value.buckets.counts.map(String),
+                    },
+                  },
+                },
+              ],
+              unit: scopeMetric.descriptor.unit || 'ms', // Default to 'ms' if no unit is specified
+            };
```

Comment:
Can this be turned into an if? I think long ternary statements like this can be hard to follow

Reply:
I used ternary here because it's more _functional_, but I see your point on readability. It's back to an if statement now.